### PR TITLE
document: add document about gcp authentication

### PR DESCRIPTION
**Pull request by YangKeao**
Signed-off-by: YangKeao <yangkeao@chunibyo.icu>

[rendered en](https://github.com/YangKeao/website-1/blob/add-gcp-authentication/docs/gcp-authentication.md)
[rendered zh](https://github.com/YangKeao/website-1/blob/add-gcp-authentication/i18n/zh/docusaurus-plugin-content-docs/current/gcp-authentication.md)

### CODE REVIEW

**Review by STRRL — COMMENTED** [review]
lgtm!

**Review by STRRL — APPROVED** [review]

File: i18n/zh/docusaurus-plugin-content-docs/current/gcp-authentication.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
+---
+title: GCP 身份验证接入
+---
+
+如果 Chaos Mesh 集群部署于 Google Cloud Platform，用户将能够通过 Google OAuth 验证登入 Chaos Dashboard。这篇文档将介绍如何配置和启用这项功能。
```
Comment (en-jin19) [inline i18n/zh/docusaurus-plugin-content-docs/current/gcp-authentication.md:5]:
```suggestion
如果 Chaos Mesh 集群部署于 Google Cloud Platform，用户将能够通过 Google OAuth 验证登入 Chaos Dashboard。本文档将介绍如何配置和启用这项功能。
```

File: i18n/zh/docusaurus-plugin-content-docs/current/gcp-authentication.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+1. 进入 [Google Cloud Platform 控制台](https://console.cloud.google.com/)
+2. 选择一个项目
+3. 在控制台的左侧菜单中选择 APIs & services
+4. 在左侧，点击 Credentials
+5. 点击 "Create Credentials"，并选择 OAuth client ID
+6. 应用类型选择 Web Application，填入合适的名称以及 Chaos Dashboard 对应的 redirect urls。Chaos Dashboard 的 redirect urls 将位于 `ROOT_URL/api/auth/gcp/callback`，其中 `ROOT_URL` 是 dashboard 的根地址，例如 `http://localhost:2333`，可以通过 `helm` 的 `dashboard.rootUrl` 配置项进行配置。
+7. 点击创建。
+
+在完成创建后，可获得该客户端的 Client ID 与 Client Secret。将它们保存下来供后续步骤使用。
```
Comment (en-jin19) [inline i18n/zh/docusaurus-plugin-content-docs/current/gcp-authentication.md:19]:
```suggestion
根据 [Setting up OAuth 2.0](https://support.google.com/cloud/answer/6158849?hl=en) 创建用于接入 GCP 的 OAuth 2.0 客户端，并获得 Client ID 与 Client Secret。

1. 进入 [Google Cloud Platform 控制台](https://console.cloud.google.com/)。
2. 选择一个项目。
3. 在控制台的左侧菜单中选择 APIs & services。
4. 点击位于左侧的 Credentials。
5. 点击 Create Credentials，并选择 OAuth client ID。
6. 应用类型选择 Web Application，填入合适的名称以及 Chaos Dashboard 对应的 redirect urls。Chaos Dashboard 的 redirect urls 将位于 `ROOT_URL/api/auth/gcp/callback`，其中 `ROOT_URL` 是 dashboard 的根地址，例如 `http://localhost:2333`，可以通过 `helm` 的 `dashboard.rootUrl` 配置项进行配置。
7. 点击创建。

完成创建后，即可获得该客户端的 Client ID 与 Client Secret，请保存这两项内容，供在后续步骤中使用。
```

File: i18n/zh/docusaurus-plugin-content-docs/current/gcp-authentication.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+## 填写配置并启动 Chaos Mesh
+
+要启动这项功能，需要打开 Chaos Mesh 的 helm charts 中的如下配置项：
```
Comment (en-jin19) [inline i18n/zh/docusaurus-plugin-content-docs/current/gcp-authentication.md:23]:
```suggestion
要启动这项功能，需要打开 Chaos Mesh 的 helm charts ，设置以下配置项：
```

File: i18n/zh/docusaurus-plugin-content-docs/current/gcp-authentication.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+1. `dashboard.gcpSecurityMode` 需要设置为 `true`
+2. `dashboard.gcpClientId` 需要设置为上一步骤中获得的 Client ID
+3. `dashboard.gcpClientSecret` 需要设置为上一步骤中获得的 Client Secret
+4. `dashboard.rootUrl` 需要设置为 Chaos Dashboard 的根地址
+
+如果已经安装并运行了 Chaos Mesh，可以通过 `helm upgrade` 命令来更新配置。而如果还未安装 Chaos Mesh，可以通过 `helm install` 来安装。
```
Comment (en-jin19) [inline i18n/zh/docusaurus-plugin-content-docs/current/gcp-authentication.md:30]:
```suggestion
如果已经安装并运行了 Chaos Mesh，可以通过 `helm upgrade` 命令来更新配置；如果还未安装 Chaos Mesh，则可以通过 `helm install` 进行安装。
```

File: i18n/zh/docusaurus-plugin-content-docs/current/gcp-authentication.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+## 开始使用
+
+打开 Chaos Dashboard，点击验证窗口下方的 Google Icon
+
+![img](./img/google-auth.png)
+
+登入 Google 账号并授予权限，即可登录 Chaos Dashboard。这时用户的权限与该 Google 账户在集群中的权限一致，可以使用 RBAC 的方式来赋予用户权限，例如：
```
Comment (en-jin19) [inline i18n/zh/docusaurus-plugin-content-docs/current/gcp-authentication.md:38]:
```suggestion
## 使用

打开 Chaos Dashboard，点击验证窗口下方的 Google Icon。

![img](./img/google-auth.png)

登入 Google 账号并授予权限后，即可登录 Chaos Dashboard。此时，用户的权限与该 Google 账户在集群中的权限一致。如需设置额外的用户权限，可以使用 RBAC 的方式来赋予其对应权限，例如：
```

File: i18n/zh/docusaurus-plugin-content-docs/current/gcp-authentication.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+subjects:
+  - kind: User
+    name: example@gmail.com
+roleRef:
+  kind: ClusterRole
+  name: chaos-mesh-cluster-manager
+  apiGroup: rbac.authorization.k8s.io
+```
+
+这一配置将赋予用户 `example@gmail.com` 浏览或创建所有类型混沌实验的权限.
```
Comment (en-jin19) [inline i18n/zh/docusaurus-plugin-content-docs/current/gcp-authentication.md:65]:
```suggestion
这一配置为用户 `example@gmail.com` 赋予了浏览或创建所有类型混沌实验的权限。
```

File: docs/gcp-authentication.md [hunk 7]
```diff
@@ -0,0 +1,65 @@
+---
+title: GCP OAuth Authentication
+---
+
+When Chaos Mesh is deployed on the Google Cloud Platform, the users can login to the Chaos Dashboard through Google OAuth. This document will describe how to enable and configure this function.
```
Comment (en-jin19) [inline docs/gcp-authentication.md:5]:
```suggestion
When Chaos Mesh is deployed on the Google Cloud Platform, you can log in to Chaos Dashboard through Google OAuth. This document describes how to enable and configure this function.
```

File: docs/gcp-authentication.md [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+
+1. Go to the Google Cloud Platform Console.
+2. From the projects list, select a project or create a new one.
+3. If the APIs & services page isn't already open, open the console left side menu and select APIs & services.
+4. On the left, click Credentials.
+5. Click New Credentials, then select OAuth client ID. 
+6. Select "Web Application" as the application type, and enter additional information and the redirect url of chaos dashboard. The redirect url of chaos dashboard is `ROOT_URL/api/auth/gcp/callback`.The `ROOT_URL` is the root url of chaos dashboard, like "http://localhost:2333". This url can be set through the `dashboard.rootUrl` field in helm config.
+7. Click Create Client ID.
+
+After creating the client, save the Client ID and Client Secret for the following steps.
```
Comment (en-jin19) [inline docs/gcp-authentication.md:19]:
```suggestion
Create GCP OAuth client and get the Client ID and Client Secret according to [Setting up OAuth 2.0](https://support.google.com/cloud/answer/6158849?hl=en).

1. Go to the [Google Cloud Platform Console](https://console.cloud.google.com/).
2. From the projects list, select a project or create a new one.
3. If the APIs & services page was not loaded automatically, open the console left side menu and select "APIs & services" manually.
4. Click "Credentials" on the left.
5. Click "New Credentials", then select "OAuth client ID". 
6. Select "Web Application" as the application type, and enter additional information and the redirect URL of Chaos dashboard, which is `ROOT_URL/api/auth/gcp/callback`. In this part, `ROOT_URL` is the root URL of Chaos dashboard, like "http://localhost:2333". This URL can be set through the `dashboard.rootUrl` field in the `helm` configuration.
7. Click "Create Client ID".

After creating the client, remember to save the Client ID and Client Secret for the following steps.
```

File: docs/gcp-authentication.md [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+## COnfigure and start the Chaos Mesh
+
+To enable this function, the users need to fill in the following fields in helm charts:
+
+1. `dashboard.gcpSecurityMode` needs to be set to `true`.
+2. `dashboard.gcpClientId` needs to be set to Client ID from the former section.
+3. `dashboard.gcpClientSecret` needs to be set to Client Secret from the former section.
+4. `dashboard.rootUrl` should be set to the root address of Chaos Dashboard.
+
+If the Chaos Mesh has been installed, the configuration could be updated through `helm upgrade`. If not, it can be installed through `helm install`.
```
Comment (en-jin19) [inline docs/gcp-authentication.md:30]:
```suggestion
## Configure and start Chaos Mesh

To enable the function, you need to set the configuration items in helm charts as follows:

1. Set `dashboard.gcpSecurityMode` to `true`.
2. Set `dashboard.gcpClientId` to the Client ID from the former section.
3. Set `dashboard.gcpClientSecret` to the Client Secret from the former section.
4. Set `dashboard.rootUrl` to the root address of Chaos Dashboard.

If Chaos Mesh has been installed, you can update the configuration items through `helm upgrade`. If not, you can install Chaos Mesh through `helm install`.
```

File: docs/gcp-authentication.md [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+
+## Start to use
+
+Open the Chaos Dashboard, click the google icon under the authentication window.
+
+![img](./img/google-auth.png)
+
+After login to the Google Account and grant permission to the oauth client, the page will be redirected to the Chaos Dashboard automatically and the user will have logged in. Then the user will have the same priviledge as the google account in this cluster, which means, it can be controlled by the RBAC mechanism. For example:
```
Comment (en-jin19) [inline docs/gcp-authentication.md:38]:
```suggestion
## Use the function

Open Chaos Dashboard, and click the google icon under the authentication window.

![img](./img/google-auth.png)

You can 

After logging in to the Google account and being granted permission to OAuth Client, the page automatically redirects to Chaos Dashboard with logged-in status. At this time, you have the same permissions as the google account in this cluster. If you need to add other permissions, you can edit the permission through the RBAC (Role-based access control). For example:
```

File: docs/gcp-authentication.md [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+subjects:
+  - kind: User
+    name: example@gmail.com
+roleRef:
+  kind: ClusterRole
+  name: chaos-mesh-cluster-manager
+  apiGroup: rbac.authorization.k8s.io
+```
+
+This configuration will enalbe the user `example@gmail.com` to read or create any chaos experiments in any namespaces.
```
Comment (en-jin19) [inline docs/gcp-authentication.md:65]:
```suggestion
By setting this configuration, the user `example@gmail.com` is enabled to see or create any chaos experiments.
```

**Comment by YangKeao** [comment]
@en-jin19 Fixed. PTAL

**Review by en-jin19 — APPROVED** [review]
LGTM

**Pull request merged**